### PR TITLE
[rearchitecture] Add functions for determining machine specs.

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This will be used by tworkers to launch batch tasks that execute utask_main. Related: https://github.com/google/clusterfuzz/issues/3008

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
@oliverchang I think there's a conflation between high end tasks in the code. I.e. high end tasks can never be preempted. Non-preemptible tasks (I assume analyze) cannot be preempted (therefore must be high end). 
Is this understanding right?

**Comment by jonathanmetzman** [comment]
TBR

**Comment by oliverchang** [comment]
> 

I don't think so. High end means different things for OSS-fuzz vs others:
- OSS-Fuzz: high end means a set of hosts that have less workers. This means more cpu/ram to dedicate for each, and was needed for blackbox JS fuzzing.
- Others/Chromium: A set of bots with higher specs.

File: src/clusterfuzz/_internal/base/tasks.py [hunk 1]
```diff
@@ -241,6 +242,51 @@ def get_utask_filters(is_chromium, is_linux):
...
+          f'Found machine template for {initial_queue_name}',
+          machine_template=template)
+      return template
+  return None
+
+
+def get_machine_templates():
+  """Returns machine templates."""
+  # TODO(metzman): Cache this.
+  clusters_config = local_config.Config(local_config.GCE_CLUSTERS_PATH).get()
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/base/tasks.py:284]: We may want to replace this with a more specialized format later, as I want us to get rid of the whole `manage_vms` bits.

**Comment by jonathanmetzman** [comment]
> > 
> 
> I don't think so. High end means different things for OSS-fuzz vs others:
> 
> * OSS-Fuzz: high end means a set of hosts that have less workers. This means more cpu/ram to dedicate for each, and was needed for blackbox JS fuzzing.
> * Others/Chromium: A set of bots with higher specs.

But this makes it seem like this is conflated: https://github.com/google/clusterfuzz/blob/aab96e6813877a307dfad7b75678fca6790cf041/src/clusterfuzz/_internal/base/tasks.py#L257

**Pull request merged**